src/windows/custom_currency.c

```diff
@@ -0,0 +1,287 @@
+            config_save_default();
+            invalidate_money_widgets();
+        }
+        break;
+
+    case WIDX_AFFIX_DROPDOWN_BUTTON:
+        gDropdownItemsFormat[0] = STR_DROPDOWN_MENU_LABEL;
+        gDropdownItemsArgs[0] = STR_PREFIX;
+
+        gDropdownItemsFormat[1] = STR_DROPDOWN_MENU_LABEL;
+        gDropdownItemsArgs[1] = STR_SUFFIX;
+
+        window_dropdown_show_text_custom_width(
+            w->x + widget->left,
+            w->y + widget->top,
+            widget->bottom - widget->top + 1,
+            w->colours[1],
+            DROPDOWN_FLAG_STAY_OPEN,
+            2,
+            widget->right - widget->left - 3
+        );
+
+        if(CurrencyDescriptors[CURRENCY_CUSTOM].affix_unicode == CURRENCY_PREFIX) {
+            dropdown_set_checked(0, true);
+        } else {
+            dropdown_set_checked(1, true);
+        }
+
+        break;
+
+    case WIDX_SYMBOL_TEXT:
+        window_text_input_raw_open( w,
+                                    WIDX_SYMBOL_TEXT,
+                                    STR_CUSTOM_CURRENCY_SYMBOL_INPUT_TITLE,
+                                    STR_CUSTOM_CURRENCY_SYMBOL_INPUT_DESC,
+                                    CurrencyDescriptors[CURRENCY_CUSTOM].symbol_unicode,
+                                    CURRENCY_SYMBOL_MAX_SIZE);
+        break;
+
+    }
+}
+
+/**
+*
+*  rct2: 0x006BB076
+*/
+static void window_custom_currency_dropdown(rct_window *w, int widgetIndex, int dropdownIndex)
+{
+    if (dropdownIndex == -1)
+            return;
+
+    if(widgetIndex == WIDX_AFFIX_DROPDOWN_BUTTON) {
+
+        if(dropdownIndex == 0) {
+            CurrencyDescriptors[CURRENCY_CUSTOM].affix_ascii = CURRENCY_PREFIX;
+            CurrencyDescriptors[CURRENCY_CUSTOM].affix_unicode = CURRENCY_PREFIX;
+        } else if(dropdownIndex == 1) {
+            CurrencyDescriptors[CURRENCY_CUSTOM].affix_ascii = CURRENCY_SUFFIX;
+            CurrencyDescriptors[CURRENCY_CUSTOM].affix_unicode = CURRENCY_SUFFIX;
+        }
+
+
+        gConfigGeneral.custom_currency_affix = CurrencyDescriptors[CURRENCY_CUSTOM].affix_unicode;
+        config_save_default();
+
+        invalidate_money_widgets();
+
+    }
+}
+
+static void window_custom_currency_text_input(struct rct_window *w, int windgetIndex, char *text)
+{
+    if(text != NULL) {
+        strncpy(CurrencyDescriptors[CURRENCY_CUSTOM].symbol_unicode,
+                text,
+                CURRENCY_SYMBOL_MAX_SIZE);
+
+        strncpy(gConfigGeneral.custom_currency_symbol,
+                CurrencyDescriptors[CURRENCY_CUSTOM].symbol_unicode,
+                CURRENCY_SYMBOL_MAX_SIZE);
+
+        config_save_default();
+        invalidate_money_widgets();
+    }
+}
+
+static void invalidate_money_widgets()
+{
+    widget_invalidate_by_class(WC_BOTTOM_TOOLBAR, 2 /*WIDX_MONEY (not accesible)*/);
+    widget_invalidate_by_class(WC_CUSTOM_CURRENCY_CONFIG, WIDX_RATE);
+}
+
+/**
+ *
+ *  rct2: 0x0066D321
```

Comment:
This is the address of the original method, but that's probably not true as this is a new feature